src/main/java/com/openlattice/controllers/OrganizationsController.java

```diff
@@ -106,15 +111,15 @@ public Organization getOrganization( @PathVariable( ID ) UUID organizationId ) {
         ensureRead( organizationId );
         //TODO: Re-visit roles within an organization being defined as roles which have read on that organization.
         Organization org = organizations.getOrganization( organizationId );
-        Set<Role> authorizedRoles = getAuthorizedRoles( organizationId, Permission.READ );
-        return new Organization(
-                org.getSecurablePrincipal(),
-                org.getAutoApprovedEmails(),
-                org.getMembers(),
-                authorizedRoles,
-                org.getApps(),
-                org.getSmsEntitySetInfo(),
-                org.getPartitions() );
+
+        Set<AclKey> authorizedRoleAclKeys = authorizations
```

Comment:
If we filter roles by the current principals having read on them in `getOrganization`, shouldn't we also do it, when calling `getRoles`?

Reply:
I think we do -- `getRoles` calls `getAuthorizedRoles` which loads the organization and filters roles to ones with READ permissions

Reply:
Then why not use it also here?

Reply:
Since this way is more efficient since it's only making one bulk authorization call here. I could rewrite that one to make more efficient calls too though...

Reply:
I see! 👍